workshop/content/lab1.5_kafka.md

```diff
@@ -0,0 +1,272 @@
+
+```
+oc get configmap config-default-kafka-channel -o yaml -n knative-eventing
+```
+
+You can see the default for `kafka` is `KafkaChannel`.  Here's the relevant snippet:
+
+```
+apiVersion: v1
+data:
+  default-ch-config: |
+    namespaceDefaults:
+      kafka:
+        apiVersion: messaging.knative.dev/v1alpha1
+        kind: KafkaChannel
+```
+
+### Create Sink
+We will now create a sink, a final destination for Kafka events flowing through the system. In this case, it will be a Serverless service. Let's create one now.
+
+First let's make sure you are still in the right project.
+
+```
+oc project
+# Using project "user-$USER_NAME" on server...
+```
+
+Now we can create our sink.
+
+```
+oc apply -f https://raw.githubusercontent.com/RedHatGov/serverless-workshop-code/main/kafka/kafka-sink.yml
+```
+
+Verify that it is there:
+
+```
+oc get ksvc
+```
+
+```
+NAME            URL                                               LATESTCREATED      LATESTREADY        READY   REASON
+eventinghello   http://eventinghello-xyz.sandbox373.opentlc.com   eventinghello-v1   eventinghello-v1   True
+```
+
+When Serverless services are created they are initially spun up, which is why we can see the logs:
+
+```
+stern eventinghello -c user-container`
+```
+
+Note, if you didn't run the `stern` command within ~90s of creating the sink, the container might have spun back down, but that's ok, you can keep going.
+
+### Create Topic
+We have a Kafka topic, we should now go ahead and create the Kafka topic.
+
+```
+curl -sS https://raw.githubusercontent.com/RedHatGov/serverless-workshop-code/main/kafka/kafka-topic.yml | sed "s/USER_NUMBER/$USER_NUMBER/" | oc apply -f -
+```
+
+Verify it was created:
+
+```
+oc get kafkatopics
+# a nice shortcut is `oc get kt`
+```
+
+```
+NAME                                                                        PARTITIONS   REPLICATION FACTOR
+my-topic-<user_number>                                                      10           1
+```
+
+### Create KafkaSource Instance
+Now we will create a KafkaSource instance that ties our topic to our sink.
+
+```
+curl -sS https://raw.githubusercontent.com/RedHatGov/serverless-workshop-code/main/kafka/kafka-source-to-sink.yml | sed "s/USER_NUMBER/$USER_NUMBER/" | oc apply -f -
+```
+
+Verify that it was created.
+
+```
+oc get kafkasource
+```
+
+```
+NAME               TOPICS                     BOOTSTRAPSERVERS                        READY   REASON   AGE
+mykafka-source     my-topic-[user-number]     my-cluster-kafka-bootstrap.kafka:9092   True             76s
+```
+
+Using the web browser, click on over to the `Developer` view.  Then click `Topology` and make sure you are in the correct `Project`.  You can then verify the KafkaSource is configured correctly.
+
+![Serverless Kafka Source](./images/serverless_kafka_source.png)
+
+### See It In Action
+Now let's see events flow through the Serverless system. Create the kafka producer again.
```

Comment:
Create the kafka producer (no again).  (I don't think we did this before at this point in the lab?)

Reply:
👍